services/venice-router/src/main/java/com/linkedin/venice/router/api/VeniceDelegateMode.java

```diff
@@ -329,6 +332,22 @@ private <H> H selectLeastLoadedHost(List<H> hosts, VenicePath path) throws Route
     }
     if (minHost == null) {
       if (path.isRetryRequest()) {
+        List<String> replicaHostIds = hosts.stream().map(x -> ((Instance) x).getNodeId()).collect(Collectors.toList());
+        LogManager.getLogger()
+            .error(
+                "VENG-10611-Debug Retry Request Aborted. Store Name: {}, Resource Name: {} ReplicaHostIds: {}, VenicePath SlowNodeSet: {}",
+                path.getStoreName(),
+                path.getResourceName(),
+                replicaHostIds,
+                path.getSlowStorageNodeSet());
+        if (path.getRetryFutureThrowable() != null) {
+          LogManager.getLogger()
```

Comment:
Can we print the retry reason here instead of in this function?
https://github.com/linkedin/venice/blob/main/services/venice-router/src/main/java/com/linkedin/venice/router/api/VeniceDelegateMode.java#L225

Reply:
Hmm but this will print out on every retry request? Instead of those who failed eventually due to slow route reason we are seeing in the incident?